app/models/workflow_step.rb

```diff
@@ -34,16 +34,25 @@ def as_milestone(xml)
                   version: version)
   end
 
-  # callback to set the completed_at column to the current time if we are completing a step
-  def set_completed_at
+  # callback to set the completed_at column to the current time if we are
+  # completing a step and completed_at is not already set
+  def maybe_set_completed
+    return unless completed?
+    return if already_completed?
```

Comment:
Can you do this as `self.completed_at ||= Time.now` ?

Reply:
done